### PR TITLE
Ensure Camel rest generated with rest endpoints even with underscored

### PR DESCRIPTION
**Pull request by apupier**
names #66

- improved test to check for it
- support underscores in names

Some tests are failing but creating to ease discussion on this issue.

[ERROR] Errors: 
[ERROR]   CamelRestDocLitTest.testRestClient:132 » NotFound HTTP 404 Not Found
[ERROR]   CamelRestRpcLitTest.testRestClient:132 » NotFound HTTP 404 Not Found

### CODE REVIEW

File: impl/src/main/java/org/jboss/fuse/wsdl2rest/impl/ResourceMapperImpl.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
     private Pattern httpPostWordsPattern = Pattern.compile(httpPostWords);
     private Pattern httpDeleteWordsPattern = Pattern.compile(httpDelWords);
     private Pattern httpPutWordsPattern = Pattern.compile(httpPutWords);
-    private Pattern resourcePattern = Pattern.compile("[a-z]+|([A-Z][a-z]+)*");
+    private Pattern resourcePattern = Pattern.compile("[A-Za-z_]+|([A-Z_][a-z_]+)*");
```
Comment (apupier) [inline impl/src/main/java/org/jboss/fuse/wsdl2rest/impl/ResourceMapperImpl.java]: I tried to hack this resource pattern as it seems that _ should be allowed (as per the wsdl provided for the reported issue)
where is the corresponding specification for this resource stuff? Can we have also other characters?
I don't understand why modifying this is breaking the tests

**Comment by tdiesler** [comment]
This fails even locally. Lets make it visible in Jenkins too ...

**Comment by tdiesler** [comment]
ok to test

**Comment by tdiesler** [comment]
Fails in Jenkins https://ci.fabric8.io/job/wsdl2rest-pullreq/18

**Comment by apupier** [comment]
> This fails even locally. Lets make it visible in Jenkins too ...

yes this is what I mentioned in the description. I created the issue to ease discussion.

**Comment by tdiesler** [comment]
A PR is a proposal for a change that must have been verified to work before the PR is created. It seems that more work is required before this issue can be resolved. As mentioned in the associated issue, you may want to create (or expand on) a comprehensive test that verifies your change.

**Pull request closed without merging**